lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -792,3 +795,31 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
 void ShaderFlags::CombineShaderFlags(const ShaderFlags &other) {
   SetShaderFlagsRaw(GetShaderFlagsRaw() | other.GetShaderFlagsRaw());
 }
+
+void ShaderFlags::GetMinimumShaderModelBasedOnFlags(unsigned &major,
```

Comment:
The introduction of this function introduces a significant and separate behavior change. We really should be doing this separately from the rest of this PR.